src/helpers/figures/charts/chart_common.ts

```diff
@@ -541,8 +541,12 @@ export function interpolateData(
       if (order === 1) {
         return predictLinearValues([values], [normalizedLabels], [normalizedNewLabels], true)[0];
       }
-      const coeffs = polynomialRegression(values, normalizedLabels, order, true).flat();
-      return normalizedNewLabels.map((v) => evaluatePolynomial(coeffs, v, order));
+      try {
+        const coeffs = polynomialRegression(values, normalizedLabels, order, true).flat();
+        return normalizedNewLabels.map((v) => evaluatePolynomial(coeffs, v, order));
+      } catch (e) {
+        return Array.from({ length: newLabels.length }, () => NaN);
+      }
```

Comment:
try/catch should be on the whole switch instead of repeated in every switch case